build.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+
+import argparse
+import contextlib
+import datetime
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import textwrap
+
+
+@contextlib.contextmanager
+def mkdtemp(*args, **kwargs):
+    tmp_dir = tempfile.mkdtemp(*args, **kwargs)
+    yield tmp_dir
+    shutil.rmtree(tmp_dir)
+
+
+def rmmkdir(dn):
+    if os.path.exists(dn):
+        shutil.rmtree(dn)
+    os.makedirs(dn)
+
+
+def git_commit_time(ref="HEAD"):
+    git_time_str = subprocess.check_output(
+        ["git", "show", "-s", "--format=%ci", ref],
+        universal_newlines=True
+    )
+    git_time_str = git_time_str.strip()
+    git_utctime = datetime.datetime.strptime(
+        git_time_str[:19] + " UTC",
+        "%Y-%m-%d %H:%M:%S %Z"
```

Comment:
For some reason on Python 2.7, `strptime` fails to parse the timezone offset correctly. So as a hack we force UTC initially. We then independently determine the offset and use this to correct the UTC time based on the time zone.